plotter/contour.go

```diff
@@ -0,0 +1,488 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:generate ./list
+
+package plotter
+
+import (
+	"image/color"
+	"math"
+	"sort"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/palette"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// Contour implements the Plotter interface, drawing
+// a contour plot of the values in the GridXYZ field. The
+// order of rows is from high index to low index down.
+type Contour struct {
+	GridXYZ GridXYZ
+
+	// Levels describes the contour heights to plot.
+	Levels []float64
+
+	// LineStyle is the style of the contour lines.
+	draw.LineStyle
+
+	// Palette is the color palette used to render
+	// the heat map.
+	Palette palette.Palette
+
+	// Underflow and Overflow are colors used to draw
```

Comment:
I see your point, but it feels to me like this is the wrong layer to put it at. A palette just defines a sequence of colors, but if we had
````
type Colormap func(float64) color.Color  // TODO: better name
````
then the warning mechanism follows through cleanly. The colormap associates z-axis numbers to colors, which could include producing warning colors at unexpected data values.